osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs

```diff
@@ -90,6 +90,15 @@ private void transitionFromResults()
                 client.ChangeState(MultiplayerUserState.Idle).FireAndForget();
         }
 
+        public override bool OnExiting(ScreenExitEvent e)
+        {
+            if (base.OnExiting(e))
+                return true;
+
+            client.LeaveRoom().FireAndForget();
```

Comment:
While we're here doing this, the xmldoc says this can throw:

https://github.com/ppy/osu/blob/f4049c7ec18fa22904e7c57ce8c725f7824136bc/osu.Game/Online/Multiplayer/IMultiplayerRoomServer.cs#L19

but looking at the server-side implementation I don't see how it can throw because this specific scenario is [guarded to do nothing instead of throwing](https://github.com/ppy/osu-server-spectator/blob/98825f65a1ac0fc333d76548139bfb4bbaf98ed2/osu.Server.Spectator/Hubs/Multiplayer/MultiplayerHub.cs#L218-L219), so might as well strip that mention.